packages/tooling/openmrs/src/commands/build.ts

```diff
@@ -33,9 +46,100 @@ function loadBuildConfig(buildConfigPath?: string): BuildConfig {
   }
 }
 
+async function extractFiles(sourceFile: string, targetDir: string) {
+  const packageRoot = "package/";
+  const rs = createReadStream(sourceFile);
+  const files = await untar(rs);
+  const packageJson = JSON.parse(
+    files[`${packageRoot}package.json`].toString("utf8")
+  );
+  const entryModule =
+    packageJson.browser ?? packageJson.module ?? packageJson.main;
+
+  Object.keys(files)
+    .filter(
+      (f) =>
+        f === "package/package.json" ||
+        f.substr(packageRoot.length) === entryModule
+    )
+    .forEach((f) => {
+      const content = files[f];
+      const fileName = f.replace(packageRoot, "");
+      const targetFile = resolve(targetDir, fileName);
+      writeFileSync(targetFile, content);
+    });
+
+  unlinkSync(sourceFile);
+}
+
+function loadCoreApps(registry: string) {
+  /*
+   * if the user specified a OMRS_ESM_CORE_APPS_DIR, we assume this has
+   * the necessary coreapps; otherwise, we check if there are any coreapps
+   * available in the places the webpack configuration will check; if there
+   * aren't, we download the apps specified in the app-shell devDependencies
+   */
+  const appsPath =
+    process.env.OMRS_ESM_CORE_APPS_DIR ??
+    resolve(require.resolve("@openmrs/esm-app-shell"), "..", "..", "apps");
+
+  let hasApp = false;
+  if (existsSync(appsPath) && statSync(appsPath).isDirectory()) {
+    hasApp = readdirSync(appsPath).some((entry) => {
+      if (statSync(entry).isDirectory()) {
+        const packageJson = resolve(entry, "package.json");
+        if (existsSync(packageJson) && statSync(packageJson).isFile()) {
+          return require(packageJson).name?.endsWith("-app");
+        }
+      }
+    });
+  }
+
+  if (!hasApp) {
+    logInfo("Fetching core apps...");
+    const devDependencies: Record<string, string> =
+      require("@openmrs/esm-app-shell/package.json").devDependencies;
+
+    const apps = Object.keys(devDependencies)
+      .filter((dep) => dep.endsWith("-app"))
+      .map((dep) => [dep, devDependencies[dep]]);
+
+    if (apps.length > 0) {
+      const cacheDir = resolve(process.cwd(), ".cache");
+      !existsSync(cacheDir) && mkdirSync(cacheDir, { recursive: true });
```

Comment:
```suggestion
      if (!existsSync(cacheDir)) {
        mkdirSync(cacheDir, { recursive: true });
      }
```

Preferable for readability. Short-circuiting is a side-effect, not the purpose of the `&&` operator.